libraries/Microsoft.Bot.Builder.Dialogs/Prompts/CloudOAuthHelper.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.OAuth;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    internal static class CloudOAuthHelper
```

Comment:
Not a fan of `*Helper` because they tend to, by design, violate single responsibility principle. Right now it looks nice but over time the name Helper does not give a lot of information to future developers / maintainers on what can be done. I'd even prefer `CloudOAuth` to `CloudOAuthHelper`.

Reply:
Agreed Helper is an awful name for anything. This class represents the prompt's abstraction around the old and new Bot Framework User Token clients - one that ties them into the TurnContext perhaps the name should just reflect that - how about "UserTokenAccess" ? (Note this is all internal so the name is quite local.)